lib/reply.js

```diff
@@ -134,16 +136,36 @@ function wrapPumpCallback (reply) {
   return function pumpCallback (err) {
     if (err) {
       reply._req.log.error(err)
-      setImmediate(wrapReplyEnd, reply, '')
+      setImmediate(handleReplyEnd, reply, '')
     }
   }
 }
 
-function wrapReplyEnd (reply, payload) {
-  reply.sent = true
```

Comment:
this can generation a lot of headache.
Why we need to move this *after* `onSend` hook?

Reply:
@delvedor

Reply:
That function closes the response, so it's the last function we must call.
Since the serialization is after the onSend hook, we have refactored a little bit how the `Reply` internals work.

Reply:
Is there a gate that doesn't allow you to call 'send' method twice?